miniwob/action.py

```diff
@@ -1,156 +1,183 @@
+    CLICK_ELEMENT = "CLICK_ELEMENT"
+    DBLCLICK_ELEMENT = "DBLCLICK_ELEMENT"
+    MOUSEDOWN_ELEMENT = "MOUSEDOWN_ELEMENT"
+    MOUSEUP_ELEMENT = "MOUSEUP_ELEMENT"
+    # Mouse wheel
+    SCROLL_UP = "SCROLL_UP"
+    SCROLL_DOWN = "SCROLL_DOWN"
+    # Keyboard
+    PRESS_KEY = "PRESS_KEY"
+    TYPE_TEXT = "TYPE_TEXT"
+    TYPE_FIELD = "TYPE_FIELD"
+    FOCUS_ELEMENT_AND_TYPE_TEXT = "FOCUS_ELEMENT_AND_TYPE_TEXT"
+    FOCUS_ELEMENT_AND_TYPE_FIELD = "FOCUS_ELEMENT_AND_TYPE_FIELD"
+
+
+COORDS_ACTIONS = {
+    ActionTypes.CLICK_COORDS,
+    ActionTypes.DBLCLICK_COORDS,
+    ActionTypes.MOUSEDOWN_COORDS,
+    ActionTypes.MOUSEUP_COORDS,
+}
+ELEMENT_ACTIONS = {
+    ActionTypes.CLICK_ELEMENT,
+    ActionTypes.DBLCLICK_ELEMENT,
+    ActionTypes.MOUSEDOWN_ELEMENT,
+    ActionTypes.MOUSEUP_ELEMENT,
+    ActionTypes.FOCUS_ELEMENT_AND_TYPE_TEXT,
+    ActionTypes.FOCUS_ELEMENT_AND_TYPE_FIELD,
+}
+TEXT_ACTIONS = {
+    ActionTypes.TYPE_TEXT,
+    ActionTypes.FOCUS_ELEMENT_AND_TYPE_TEXT,
+}
+FIELD_ACTIONS = {
+    ActionTypes.TYPE_FIELD,
+    ActionTypes.FOCUS_ELEMENT_AND_TYPE_FIELD,
+}
+
+
+@dataclass
+class ActionSpaceConfig:
+    """Configurations for the action space.
+
+    Attributes:
+        action_types: An ordered sequence of action types to include.
+            The order will be used for interpreting the Discrete space.
+        screen_width: Screen width. Will be overridden by MiniWoBEnvironment.
+        screen_height: Screen height. Will be overridden by MiniWoBEnvironment.
+        coord_bins: If specified, bin the x and y coordinates to these numbers
+            of bins. Mouse actions will be executed at the middle of the
+            specified partition.
+        allowed_keys: An ordered sequence of allowed keys and key combinations
+            for the PRESS_KEY action. The order will be used for interpreting
+            the Discrete space.
+        text_max_len: Maximum text length for the TYPE_TEXT action.
+        text_charset: Character set for the TYPE_TEXT action.
+    """
+
+    action_types: Sequence[ActionTypes]
+    screen_width: Optional[float] = None
+    screen_height: Optional[float] = None
+    coord_bins: Optional[Tuple[int, int]] = None
+    allowed_keys: Sequence[str] = DEFAULT_ALLOWED_KEYS
+    text_max_len: int = TYPING_MAX_LENGTH
+    text_charset: Union[str, Set[str]] = ASCII_CHARSET
+
+    @classmethod
+    def get_preset(cls, name="all_supported"):
+        """Returns a preset config."""
+        if name == "all_supported":
+            return cls(
+                action_types=[
+                    ActionTypes.NONE,
+                    ActionTypes.CLICK_COORDS,
+                    ActionTypes.CLICK_ELEMENT,
+                    ActionTypes.TYPE_TEXT,
+                    ActionTypes.FOCUS_ELEMENT_AND_TYPE_TEXT,
+                ]
+            )
+        else:
+            raise ValueError(f"Unknown preset name {name}")
+
+
+def get_action_space(config: ActionSpaceConfig) -> spaces.Space:
```

Comment:
Do we want `get_action_space` and `get_raw_coords` as `ActionSpaceConfig` methods

Reply:
I have moved the methods to ActionSpaceConfig.